src/Builder.js

```diff
@@ -34,6 +67,9 @@ Builder.prototype = {
     var terms = [];
     for (var idx = 0; idx < arguments.length; idx++) {
       var arg = arguments[idx];
+      if (!(arg instanceof pexprs.PExpr)) {
```

Comment:
I was thinking that everything of the form [ctorName, ...] could be recursively traversed, and constructed in a bottom-up manner. E.g.:

```js
function fromRecipe(r) {
  if (!Array.isArray(r)) {
    return r;  // Strings, Numbers, etc. don't need further processing.
  }
  var ctorName = r[0];
  var args = r.slice(2);
  var result = ctorName.apply(this, args.map(fromRecipe));
  ...
  return result;
}
```

This way, the individual constructors (`plus`, `opt`, `star`, etc.) wouldn't have to call `fromRecipe`. Of course this wouldn't work for cases where a constructor takes an Array as an argument. And maybe it doesn't work for other reasons -- you've thought about this more. Just wanted to clarify what I was trying to suggest.